### PR TITLE
Lab2 luliang li

### LINKED ISSUE #268: Lab 2 Bubble sort
- [ ] Create an executable go program in directory `02_bubble/luliangLi`
- [ ] Write a function that returns a sorted copy of `int` slice `s` using [bubble sort](https://en.wikipedia.org/wiki/Bubble_sort):
```
func bubble(s []int) []int
```
Call `fmt.Println(bubble([]int{3, 2, 1, 5}))` in `main` to print:
```
[1 2 3 5]
```
- [ ] Bonus points: implement [Insertion sort](https://en.wikipedia.org/wiki/Insertion_sort)

### PR DESCRIPTION
**Pull request by luliangLi**
Fixes #268 

Review of colleague's PR #182 

#### Changes proposed in this PR:
- implemented bubble/insert sort 
- implemented test cases

### CODE REVIEW

File: 02_bubble/luliangLi/main_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	}{
+		"normal":     {input: []int{3, 2, 1, 5}, want: []int{1, 2, 3, 5}},
+		"blank":      {input: []int{}, want: []int{}},
+		"nil":        {input: nil, want: nil},
+		"duplicated": {input: []int{1, 0, 0, 1}, want: []int{0, 0, 1, 1}},
+	}
+
+	for name, test := range testCases {
+		// t.Run creates a sub test and runs it like a normal test
+		test := test
```
Comment (eastrd) [inline 02_bubble/luliangLi/main_test.go:25]: This line is extra

File: 02_bubble/luliangLi/main_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
```
Comment (eastrd) [inline 02_bubble/luliangLi/main_test.go:56]: Same as above, this line is not needed

**Review by eastrd — DISMISSED** [review]
Maybe add an extra test case for array length of 1 ?

File: 02_bubble/luliangLi/main_test.go (hunk 1 above)
Comment (luliangLi) [reply 02_bubble/luliangLi/main_test.go:25]: Thanks and please refer to my answer in #177

**Review by anzboi — DISMISSED** [review]
Perfect as far as I can tell. LGTM 👍

**Review by anzboi — DISMISSED** [review]

**Comment by anzboi** [comment]
Apologies for the confusion, You need to produce a copy of the original array. As it is, your solution modifies the original array. Can you please change this and write a test to ensure the original array is not modified.

**Review by anzboi — APPROVED** [review]
LGTM.
For future labs we would ask that you use more detailed commit messages.

**Comment by juliaogris** [comment]
The go-course is now closed. Thank you very much for participating.

**Pull request closed without merging**